parts/linux/cloud-init/artifacts/cse_start.sh

```diff
@@ -49,6 +49,16 @@ JSON_STRING=$( jq -n \
 mkdir -p /var/log/azure/aks
 echo $JSON_STRING | tee /var/log/azure/aks/provision.json
 
+# Create stage marker for two-stage workflow
+if [ "${PRE_PROVISION_ONLY}" = "true" ]; then
+    # Stage 1: Create marker indicating Stage 2 is needed
```

Comment:
Please document it somewhere what are done (high level) in stage 1 & 2 and why we need it for pre-provisioning mode. Also under what scenarios we will need pre-provisioning, etc. Thanks!